runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -49,6 +53,13 @@
 
   private static final EnvironmentIdExtractor DEFAULT_SPEC_EXTRACTOR = (transform) -> null;
 
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .registerModules(ObjectMapper.findModules(ReflectHelpers.findClassLoader()));
+  public static final String ENVIRONMENT_DOCKER = "DOCKER";
+  public static final String ENVIRONMENT_PROCESS = "PROCESS";
+  public static final String ENVIRONMENT_EMBEDDED = "EMBEDDED"; // Non Public urn for testing
```

Comment:
It is used at other places like FlinkDefaultExecutableStageContext so changing the visibility will not work.

Reply:
This should be in the proto (already discussed on the original ML thread). It is not only useful for testing but will in the future also be an option to execute the Java SDK in the JVM based runner.

Reply:
EMBEDDED requires more discussion before addition to standard environments. I would prefer to avoid its addition before that gets resolved.
Referece: https://github.com/apache/beam/pull/6373#discussion_r217828345

Reply:
That discussion took place here: https://lists.apache.org/thread.html/d8b81e9f74f77d74c8b883cda80fa48efdcaf6ac2ad313c4fe68795a@%3Cdev.beam.apache.org%3E

I'm fine leaving this for follow-up as long as no extra effort is spent working around the issue that the option is not in the proto.

Reply:
Thanks for the link.
Yes, the addition should not take much effort.
I add it in a followup PR